### ISSUE TITLE
react/hook-use-state not working with camelcase

### ISSUE THREAD

**Issue by jakeleventhal** [yannickcr/eslint-plugin-react#3243]
The following code should be acceptable; however, I get an error saying "useState call is not restructured into value + setter pair  `react/hook-use-state`"
```javascript
const [btcValue, setBTCValue] = useState<boolean>(false);
```
It should be acceptable because BTC is an all caps abbreviation. It is not an exact string match with `btcValue` but it doesn't follow camelcase to write `setbtcValue`

**Comment by ljharb** [comment]
I agree; java-style camelcasing algorithms would have you use `setBtcValue`, but that's atrocious.

This should allow all-caps versions of a state segment in the setter, to account for initialisms. cc @duncanbeevers
